src/wallet/wallet.cpp

```diff
@@ -2902,23 +2902,6 @@ OutputType CWallet::TransactionChangeType(OutputType change_type, const std::vec
         return change_type;
     }
 
-    // if m_default_address_type is legacy, use legacy address as change (even
```

Comment:
I'd prefer factoring out this change from the "Change default address to bech32" commit, because it doesn't seem directly related, but maybe I'm wrong.

Reply:
IIRC mostly just I'm breaking tests twice and didn't want to bother, let me see if I swap around my commits if it makes it less stupid...

Reply:
ok, split it up with passing tests for each